vmaas_sync/metrics_cyndi.go

```diff
@@ -0,0 +1,125 @@
+package vmaas_sync //nolint:golint,stylecheck
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"fmt"
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+)
+
+const (
+	allSystemCount        = "allSystem"
+	uniqueTagsCount       = "uniqueTags"
+	systemsSapSystemCount = "systemsSapSystem"
+	systemsWithTagsCount  = "systemsWithTag"
+)
+
+var (
+	cyndiSystemsCnt = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "patchman_engine",
+		Subsystem: "vmaas_sync",
+		Name:      "cyndi_systems",
+	}, []string{"type"})
+
+	cyndiTagsCnt = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "patchman_engine",
+		Subsystem: "vmaas_sync",
+		Name:      "cyndi_tags_count",
+	}, []string{"type"})
+
+	tagsPerSystem = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: "patchman_engine",
+		Subsystem: "vmaas_sync",
+		Name:      "tags_per_system",
+	})
+)
+
+type InventoryHosts struct {
+	SystemsCount int
+	UniqueTags   int
+	SapCount     int
+	TagsCount    int
+}
+
+func updateCyndiData() {
+	stats, err := getCyndiData()
+	if err != nil {
+		utils.Log("err", err.Error()).Error("unable to update cyndi metrics")
+		stats = InventoryHosts{}
+	}
+	cyndiTagsCnt.WithLabelValues(allSystemCount).Set(float64(stats.SystemsCount))
+	cyndiTagsCnt.WithLabelValues(uniqueTagsCount).Set(float64(stats.UniqueTags))
+	cyndiTagsCnt.WithLabelValues(systemsSapSystemCount).Set(float64(stats.SapCount))
+	cyndiTagsCnt.WithLabelValues(systemsWithTagsCount).Set(float64(stats.TagsCount))
+}
+
+func getCyndiData() (stats InventoryHosts, err error) {
+	err = database.Db.Table("inventory.hosts").
+		Select("COUNT(*) as systems_count, (SELECT COUNT(*) FROM inventory.hosts WHERE" +
```

Comment:
Oh, that's not to easy to read. Is it possible to use gorm style for that? I mean something like:
~~~go
var nSAPSystems int
err := database.Db.Table("inventory.hosts").Where("system_profile -> 'sap_system' = 'true'").Count(&nSAPSystems).Error
// ...
~~~
You don't need to do it using single query. Rather split it to keep it more clear.